config/initializers/rabl_init.rb

```diff
@@ -35,9 +35,37 @@ def data_name(data_token)
       # otherwise return super since v1 has object root (config.include_child_root = true)
       super
     end
+
+    def template_extensions_from_plugins
+      return [] unless @_options[:template]
+      Foreman::Plugin.all.flat_map {|plugin| plugin.api_template_extensions(@_options[:template]) }
+    end
+
+    def load_extensions_from_plugins
+      template_extensions_from_plugins.each do |extension|
+        extends extension unless @_settings[:extends].map { |e| e[:file] }.include?(extension)
+      end
+    end
+
+    def render(*args, &block)
+      load_extensions_from_plugins
+      super
+    end
+  end
+
+  module RablTemplateHandlerExt
+    def call(template)
+      source = template.source
+
+      %{ ::Rabl::Engine.new(#{source.inspect}, :template => '#{template.virtual_path}').
+                                          apply(self, assigns.merge(local_assigns)).
+                                                        render }
+    end
   end
 end
 Rabl::Engine.send(:prepend, Foreman::RablEngineExt)
+ActionView::Template::Handlers::Rabl.singleton_class.send(:prepend, Foreman::RablTemplateHandlerExt)
```

Comment:
Did you consider using RABL's `partial` call as I do [with facets](https://github.com/theforeman/foreman/blob/4d563b215b72081bb7b63f292bd297438192e462/app/views/api/v2/hosts/main.json.rabl#L56)?

Reply:
I believe we would get similar results. What do you think could be gained from this?

Reply:
Easier to read, less monkey patching?

Reply:
I see that we already overload the `Engine` class anyway.